### PR TITLE
`pageTemplate`

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR merges `frontTemplate` and `articleTemplate` into one file, `pageTemplate`

## Why?
Because we want DCR to support more types of page and having a single abstraction for setting up a page allows us to centralise some important logic, making it easier to have more page types

### Before
```
/Article --> Article.tsx --> articleToHtml --> articleTemplate.ts
                                              
/Front --> FrontPage.tsx --> frontToHtml --> frontTemplate.ts
```

### After
```
/Article --> Article.tsx --> articleToHtml
                                              --> pageTemplate.ts
/Front --> FrontPage.tsx --> frontToHtml
```

### CODE REVIEW

File: dotcom-rendering/src/web/server/pageTemplate.ts [hunk 1]
```diff
@@ -172,20 +193,24 @@ https://workforus.theguardian.com/careers/product-engineering/
...
+						: ''
+				}
 
                 <!-- TODO make this conditional when we support more content types -->
                 ${ampLink ? `<link rel="amphtml" href="${ampLink}">` : ''}
 
                 ${fontPreloadTags.join('\n')}
 
-                ${openGraphMetaTags}
+                ${openGraphMetaTags ?? ''}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/server/pageTemplate.ts:209]: Do we want to make this explicit?

```suggestion
                ${openGraphMetaTags ?? '<!-- no Open Graph meta tags -->'}
```

File: dotcom-rendering/src/web/server/pageTemplate.ts [hunk 2]
```diff
@@ -274,13 +299,20 @@ https://workforus.theguardian.com/careers/product-engineering/
...
+                    <img src="https://sb.scorecardresearch.com/p?${new URLSearchParams(
+						{
+							c1: '2',
+							c2: '6035250',
+							cv: '2.0',
+							cj: '1',
+							cs_ucfr: '0',
+							comscorekw: keywords,
+						},
+					).toString()}" />
```
Comment (mxdvl) [inline dotcom-rendering/src/web/server/pageTemplate.ts:315]: 😍

File: dotcom-rendering/src/web/components/Article.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ type Props = {
 
 /**
  * @description
- * Article is a high level wrapper for pages on Dotcom. Sets strict mode and some globals
+ * Article is a high level wrapper for article pages on Dotcom. Sets strict mode and some globals
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Article.tsx:28]: For consistency I would rename this file.

```suggestion
 * ArticlePage is a high level wrapper for article pages on Dotcom. Sets strict mode and some globals
```

**Review by mxdvl — APPROVED** [review]
I like it, knowing that the two templates have kept in lockstep since the beginning of the fronts migration gives me confidence that this is the right abstraction.

File: dotcom-rendering/src/web/server/pageTemplate.ts (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/server/pageTemplate.ts:209]: Good idea. Done 👍

File: dotcom-rendering/src/web/components/Article.tsx (hunk 3 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Article.tsx:28]: Yeah, this makes a lot of sense. Implemented

File: dotcom-rendering/src/web/server/pageTemplate.ts (hunk 2 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/server/pageTemplate.ts:319]: I thought this version was much nicer

Comment (mxdvl) [reply dotcom-rendering/src/web/server/pageTemplate.ts:319]: I had been meaning to refactor it for a while and this is a lot cleaner.

File: dotcom-rendering/src/web/server/pageTemplate.ts [hunk 4]
```diff
@@ -172,20 +193,28 @@ https://workforus.theguardian.com/careers/product-engineering/
...
+						: '<!-- no linked data -->'
+				}
 
                 <!-- TODO make this conditional when we support more content types -->
-                ${ampLink ? `<link rel="amphtml" href="${ampLink}">` : ''}
+                ${
+					ampLink
+						? `<link rel="amphtml" href="${ampLink}">`
+						: '<!-- no Amp link -->'
+				}
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/server/pageTemplate.ts:209]: I'm a bit late but I _love_ this change to add comments when something isn't added to the page - can definitely see this coming in handy some day!

**Comment by oliverlloyd** [comment]
This work relates to https://github.com/guardian/dotcom-rendering/issues/5881

**Pull request merged**